File: Polimorfismo/Ejercicio2/Python/Figuras.py
class Cuadrado:
    def __init__(self, lado):
        self.lado = lado
    def area(self):
        return self.lado ** 2
    def mostrar(self):
        print("Cuadrado: ", self.lado)
        print(self.area())
    
class Rectangulo:
    def __init__(self, altura, base):
        self.altura = altura
        self.base = base
    def area(self):
        return self.altura * self.base
    def mostrar(self):
        print("Rectangulo: ", self.altura, self.base)
        print(self.area())
    
class Circulo:
    def __init__(self, radio):
        self.radio = radio
    def area(self):
        return 3.14 * self.radio ** 2
    def mostrar(self):
        print("Circulo: ", self.radio)
        print(self.area())
cuadrado1 = Cuadrado(2)
rectangulo1 = Rectangulo(3, 4)
circulo1 = Circulo(5)
cuadrado1.mostrar()
rectangulo1.mostrar()
circulo1.mostrar()

File: Polimorfismo/Ejercicio2/Java/Figuras.java
class Cuadrado {
    private int lado;
    public Cuadrado(int lado) {
        this.lado = lado;
    }
    public int area() {
        return lado * lado;
    }
    public void mostrar() {
        System.out.println("Cuadrado: " + lado);
        System.out.println("El area es: " + area());
    }
}
class Rectangulo {
    private int base;
    private int altura;
    public Rectangulo(int base, int altura) {
        this.base = base;
        this.altura = altura;
    }
    public int area() {
        return base * altura;
    }
    public void mostrar() {
        System.out.println("Rectángulo: " + base + " " + altura);
        System.out.println("El area es: " + area());
    }
}
class Circulo {
    private int radio;
    public Circulo(int radio) {
        this.radio = radio;
    }
    public double area() {
        return Math.PI * radio * radio;
    }

    public void mostrar() {
        System.out.printf("Círculo: " + radio + "\n");
        System.out.printf("El area es: " + area());
    }
}